### DISCUSSION TITLE
Is there any way to reduce the time to register sharding when the bot starts?

### DISCUSSION THREAD

**Discussion by nttstar** [Rapptz/discord.py#9543] [Q&A]
When I set a relatively large shard_count on a single server, such as 64/128 and so on. The bot takes about 5-10 minutes to start, is there a way to reduce this time? The premise is that the shard_count of a single machine cannot be reduced.

**Comment by Rapptz** [comment]
Unless you're being opted into a higher `max_concurrency` value (which, right now the library does not use due to it requiring synchronisation anyway...) you're limited in the API to 1 IDENTIFY every 5 seconds. 64 * 5 seconds = 320 seconds = 5 minutes and 20 seconds, so this makes sense to me.

**Comment by nttstar** [comment]
@Rapptz Thanks for your reply! So there is no other way but to increase the number of servers (reduce the number of sharding per server), right?

**Reply by Rapptz** [reply]
This limitation exists regardless of number of processes. You need Discord to give you a higher `max_concurrency` to break this limit, but this is limited to large bots over like 250k or 150k guilds. I do not remember off hand.

**Reply by nttstar** [reply]
In fact, I exceed 250,000.. And if I applied for a higher **max_concurrency**, how can I apply it in discord.py?

**Reply by Rapptz** [reply]
The library doesn't use it because at that scale it's kind of expected that you'll do the synchronisation yourself since it involves orchestrating multiple processes. I want to get around to it... eventually at some point but it's not a very high priority item. You can track this here: https://github.com/Rapptz/discord.py/issues/8430

**Reply by nttstar** [reply]
Thanks!
